core/simplerouter.go

```diff
@@ -35,7 +45,13 @@ func (router *SimpleRouter) Configure(conf PluginConfigReader) error {
 	router.Log = conf.GetLogScope()
 	router.Timeout = nil
 	router.streamID = conf.GetStreamID("Stream", GetStreamID(conf.GetID()))
-	router.modulators = conf.GetModulatorArray("Modulators", router.Log, ModulatorArray{})
+
+	// init router.modulators with filters only
+	filters := conf.GetFilterArray("Filters", router.Log, FilterArray{})
+	for _, filter := range filters {
+		filterModulator := NewFilterModulator(filter)
```

Comment:
Summary of our internal discussion:
The Filter interface is not recursive. This happens because the return value leaves it to the caller on who to react on a "reject". This means the caller needs to get the DropStreamID from the Filter. An array cannot easily implement GetDropStreamID as in case of a reject the DropStreamID is bound to a specific index for that specific call.
We have several options to fix this:
1. Let filter return the drop stream id. This puts additional burden on the maintainer of the filter (return the correct config value in case of reject)
2. Write a GetDropStreamID for FilterArray that re-evaluates alle filters. This collides with filters that store internal counters like the rate limiter.

As of this we decided to leave it as is and do the "enrichment" on the next higher level, i.e. the modulator. This is currently the easiest implementation but is still bad in terms of type safety and readability.

I.e.: This is an architectural TBD and might change with a later PR